api/client/inventory.go

```diff
@@ -0,0 +1,511 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"io"
+	"sync"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/trace"
+	"github.com/gravitational/trace/trail"
+)
+
+// DownstreamInventoryControlStream is the client/agent side of a bidirectional stream established
+// between teleport instances and auth servers.
+type DownstreamInventoryControlStream interface {
```

Comment:
I won't block the PR on this, but this inventory.go file seems like it should be pretty easy to write some unit tests to verify the basic functionality (start recv loops, send a ping, close and wait for termination, etc)

Reply:
Added test in `lib/auth` that sets up a client and performs basic hello exchange + ping.